<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>try-catch문</title>
</head>
<body>
    <h1>
         예외처리 프로세스 : 예상되는 에러를 try{code..} 영역에 배치하고, 에러가 발생하면
         catch(error){code...}영역에서 처리
    </h1>
    <ul>
        <li>
            try영역: 에러발생이 예상되는 코드를 실행하는 곳.
        </li>
        <li>
            catch(err)영역: 에러 발생시 예외처리를 실행하는 곳
        </li>
        <li>
            try~catch문으로 예외처리를 하면 프로세스를 에러로 인해 멈추지 않고 지속가능 
        </li>
        <li>
            finally영역: 에러 유무와 상관없이 무조건 실행되야 하는 영역
        </li>
        <li>
            throw: 예외를 고의적으로 발생시키는 키워드,
        </li>
        <li>
            throw + 예외발생 후 실행되는 코드
        </li>
    </ul>
    <script>
        console.log('프로세스 시작');
        try{
            alerttt('경고창 실행');
            console.log('경과창 실행확인')
        }catch(err){
            console.log('catch영역');
            console.log(err.name);
            console.log(err.message);
            try{
                if(err.message.toString().length > 0){ //error메시지의 값이 있다면
                    throw 'throw error';
                }
            }catch(err2){
                console.log('err2',err2.message);
            }finally{
                console.log('err2','finally.....');
            }
        }finally{
            console.log('err','finally... 무조건 실행되어야 하는 영역');
        }
        console.log('프로세스 종료');
    </script>
</body>
</html>